codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -213,6 +214,191 @@ private String generateHttpBindingSetter(Shape targetShape, Symbol targetSymbol,
+                return "strConv.ParseInt(" + operand + ", 0, 0 )";
+            case BLOB:
+                writer.addUseImports(base64EncPackageSymbol);
+                writer.write("b, err := base64.StdEncoding.DecodeString($L)", operand);
+                writer.write("if err != nil { return err }");
+                return "b";
+            case STRUCTURE:
+                // Todo: delegate to the shape deserializer
+                break;
+            default:
+                throw new CodegenException("unexpected shape type " + targetShape.getType());
+        }
+        return value;
+    }
+
+    private void writeRestDeserializerMember(
+            GoWriter writer,
+            Model model,
+            SymbolProvider symbolProvider,
+            HttpBinding binding
+    ) {
+        MemberShape memberShape = binding.getMember();
+        Shape targetShape = model.expectShape(memberShape.getTarget());
+        Symbol targetSymbol = symbolProvider.toSymbol(targetShape);
+        String memberName = symbolProvider.toMemberName(memberShape);
+        String bindingLocation = binding.getLocationName();
+
+        switch (binding.getLocation()) {
+            case HEADER:
+                writer.openBlock("if val := response.Header.Get($S); val != $S {", "}",
+                        bindingLocation, "", () -> {
+                    boolean pointable = targetSymbol.getProperty("pointable", Boolean.class).orElse(false);
+                    String value = generateHttpBindingsValue(targetShape, targetSymbol, writer, "val");
+                    if (pointable) {
+                        value = "&" + value;
+                    }
+                    writer.write("v.$L = $L", memberName, value);
+                });
+                break;
+            case PREFIX_HEADERS:
+                String prefix = binding.getLocationName();
+                if (targetShape.isMapShape()) {
+                    Shape targetValueShape = model.expectShape(targetShape.asMapShape().get().getValue().getTarget());
+                    Symbol targetValueSymbol = symbolProvider.toSymbol(targetValueShape);
+                    for (Shape shape: targetShape.asMapShape().get().members()) {
+                        String name = shape.getId().getName();
+                        String locationName = prefix + name;
+                        writer.openBlock(
+                                "if val := response.Header.Get($S); val != $S {",
+                                "}", locationName, "", () -> {
+                                    writer.write("v.$L[$L] = $L",
+                                            memberName, name,
+                                            generateHttpBindingsValue(targetValueShape, targetValueSymbol, writer,
+                                                    "val"));
+                                });
+                    }
+                } else if (targetShape.isStructureShape()) {
```

Comment:
The [prefix headers trait](https://awslabs.github.io/smithy/1.0/spec/core/http-traits.html#httpprefixheaders-trait) cannot target a structure.